imap_processing/ena_maps/utils/spatial_utils.py

```diff
@@ -0,0 +1,270 @@
+    """
+
+    def __init__(
+        self,
+        spacing_deg: float = 0.5,
+        centered_azimuth: bool = False,
+        centered_elevation: bool = True,
+        reversed_elevation: bool = False,
+    ) -> None:
+        # Store grid properties
+        self.centered_azimuth = centered_azimuth
+        self.centered_elevation = centered_elevation
+        self.reversed_elevation = reversed_elevation
+
+        # Internally, work in radians, regardless of desired output units
+        # If angular_units == deg, conversion will be done at the end
+        self.spacing = np.deg2rad(spacing_deg)
+
+        # Ensure valid grid spacing (positive, divides evenly into pi radians)
+        if self.spacing <= 0:
+            raise ValueError("Spacing must be positive valued, non-zero.")
+
+        if not np.isclose((np.pi / self.spacing) % 1, 0):
+            raise ValueError("Spacing must divide evenly into pi radians.")
+
+        # build_spacial_bins creates the bin edges and centers for azimuth and elevation
+        # E.g. for spacing=1, az_bin_edges = [0, 1, 2, ..., 359, 360] deg.
+        (az_bin_edges, el_bin_edges, az_bin_midpoints, el_bin_midpoints) = (
+            build_spatial_bins(az_spacing=spacing_deg, el_spacing=spacing_deg)
+        )
+
+        # Store the bin edges and midpoints in radians
+        self.az_bin_edges = np.deg2rad(az_bin_edges)
+        self.el_bin_edges = np.deg2rad(el_bin_edges)
+        self.az_bin_midpoints = np.deg2rad(az_bin_midpoints)
+        self.el_bin_midpoints = np.deg2rad(el_bin_midpoints)
+
+        # By default, build_spacial_bins creates bins from az=0->360 and el=-90->90.
+        if centered_azimuth:
+            self.az_bin_midpoints = self.az_bin_midpoints - np.pi
+        if not centered_elevation:
+            self.el_bin_midpoints = self.el_bin_midpoints + np.pi / 2
+
+        # If desired, reverse the elevation range so that the grid is in the order
+        # defined by the Ultra prototype code (`build_dps_grid.m`).
+        if self.reversed_elevation:
+            self.el_bin_midpoints = self.el_bin_midpoints[::-1]
+            self.el_bin_edges = self.el_bin_edges[::-1]
+
+        # Deriving our az/el grids with indexing "ij" allows for ravel_multi_index
+        # to work correctly with 1D digitized indices in each az and el,
+        # using the same ravel order ('C' or 'F') as the grid points were unwrapped.
+        self.az_grid, self.el_grid = np.meshgrid(
+            self.az_bin_midpoints, self.el_bin_midpoints, indexing="ij"
+        )
+
+        # Keep track of number of points on the grid
+        self.grid_shape = self.az_grid.shape
+        self.grid_size = self.az_grid.size
+
+        # Create degree property attributes for all radian attributes
+        self.__init_properties__()
+
+    def __init_properties__(self) -> None:
+        """Automatically generate degree properties for all radian attributes."""
```

Comment:
This is a neat way to do this :)

I am worried about adding this just for the sake of adding it. I really find that swapping between degrees/radians leads to confusion and more options can be worse for the end user/easier to make a small mistake later on. So if we can work with only radians in all inputs/outputs of your helper functions etc, and only have the entrypoint to create a map in degrees or something like that it might make our lives easier. `create_my_map(degrees_inputs) -> create_bins(radians) -> more_radians_calls()`